common/namespace/registry.go

```diff
@@ -0,0 +1,493 @@
+
+	ns, err := r.getNamespaceByID(id)
+	if err != nil {
+		return "", err
+	}
+	return ns.Name(), nil
+}
+
+func (r *registry) Refresh() {
+	replyCh := make(chan struct{})
+	r.triggerRefreshCh <- replyCh
+	<-replyCh
+}
+
+func (r *registry) refreshLoop(ctx context.Context) error {
+	timer := time.NewTicker(CacheRefreshInterval)
+	defer timer.Stop()
+
+	// Put timer events on our channel so we can select on just one below.
+	go func() {
+		for range timer.C {
+			select {
+			case r.triggerRefreshCh <- nil:
+			default:
+			}
+
+		}
+	}()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case replyCh := <-r.triggerRefreshCh:
+			for err := r.refreshNamespaces(ctx); err != nil; err = r.refreshNamespaces(ctx) {
+				select {
+				case <-ctx.Done():
+					return nil
+				default:
+					r.logger.Error("Error refreshing namespace cache", tag.Error(err))
+					select {
+					case <-time.After(CacheRefreshFailureRetryInterval):
+					case <-ctx.Done():
+						return nil
+					}
+				}
+			}
+			if replyCh != nil {
+				replyCh <- struct{}{}
+			}
+		}
+	}
+}
+
+func (r *registry) refreshNamespaces(ctx context.Context) error {
+	now := r.timeSource.Now()
+
+	// first load the metadata record, then load namespaces
+	// this can guarantee that namespaces in the cache are not updated more than metadata record
+	metadata, err := r.persistence.GetMetadata()
+	if err != nil {
+		return err
+	}
+	namespaceNotificationVersion := metadata.NotificationVersion
+
+	var token []byte
+	request := &persistence.ListNamespacesRequest{PageSize: cacheRefreshPageSize}
+	var namespaces Namespaces
+	continuePage := true
+
+	for continuePage {
+		request.NextPageToken = token
+		response, err := r.persistence.ListNamespaces(request)
+		if err != nil {
+			return err
+		}
+		token = response.NextPageToken
+		for _, namespace := range response.Namespaces {
+			namespaces = append(namespaces, FromPersistentState(namespace))
+		}
+		continuePage = len(token) != 0
+	}
+
+	// we mush apply the namespace change by order
+	// since history shard have to update the shard info
+	// with namespace change version.
+	sort.Sort(namespaces)
+
+	var prevEntries []*Namespace
+	var nextEntries []*Namespace
+
+	// make a copy of the existing namespace cache, so we can calculate diff and do compare and swap
+	newCacheNameToID := cache.New(cacheMaxSize, &cacheOpts)
+	newCacheByID := cache.New(cacheMaxSize, &cacheOpts)
+	for _, namespace := range r.getAllNamespace() {
+		newCacheNameToID.Put(namespace.info.Name, namespace.info.Id)
+		newCacheByID.Put(namespace.info.Id, namespace)
+	}
+
+UpdateLoop:
+	for _, namespace := range namespaces {
+		if namespace.notificationVersion >= namespaceNotificationVersion {
+			// this guarantee that namespace change events before the
+			// namespaceNotificationVersion is loaded into the cache.
+
+			// the namespace change events after the namespaceNotificationVersion
+			// will be loaded into cache in the next refresh
+			break UpdateLoop
```

Comment:
Understand that this is not changed in this PR. But I think this should be continue, not break? Through eventually, it may not matter.

Reply:
I also don't understand why break is used here. But I'm trying to minimize non-essential changes. And the core of this particular PR is the renaming.

Reply:
I would not mix any logic change with big rename PR like this one.